ui/app/models/kv/metadata.js

```diff
@@ -67,10 +68,16 @@ export default class KvSecretMetadataModel extends Model {
     return keyIsFolder(this.path);
   }
 
+  // cannot use isDeleted due to ember property conflict
+  get isSecretDeleted() {
+    return isDeleted(this.deletionTime);
+  }
+
   // turns version object into an array for version dropdown menu
   get sortedVersions() {
     const array = [];
     for (const key in this.versions) {
+      this.versions[key].isSecretDeleted = isDeleted(this.versions[key].deletion_time);
```

Comment:
We shouldn't need to set `isSecretDeleted` here since it's on the model itself, right?

Reply:
Sorry ignore this -- I just realized this is on the metadata response

Reply:
I thought the same, but then did a little debugging. this.versions, is just the version chunk of the endpoint. And the way we consumed this getter we check version status.

```
"versions": {
      "1": {
        "created_time": "2018-03-22T02:24:06.945319214Z",
        "deletion_time": "",
        "destroyed": false
      },
      "2": {
        "created_time": "2018-03-22T02:36:33.954880664Z",
        "deletion_time": "",
        "destroyed": false
      },
      "3": {
        "created_time": "2018-03-22T02:36:43.986212308Z",
        "deletion_time": "",
        "destroyed": false
      }
```

Example of this getter being consumed in the app.
```
        {{#each @metadata.sortedVersions as |versionData|}}
....
            {{else if versionData.isSecretDeleted}}
                <Icon @name="x-square-fill" class="has-text-grey is-pulled-right" />
            </LinkTo>
....
          </li>
        {{/each}}
```